src/Libraries/Microsoft.Extensions.AI/ChatCompletion/FunctionInvokingChatClient.cs

```diff
@@ -62,12 +62,14 @@ public partial class FunctionInvokingChatClient : DelegatingChatClient
     /// Initializes a new instance of the <see cref="FunctionInvokingChatClient"/> class.
     /// </summary>
     /// <param name="innerClient">The underlying <see cref="IChatClient"/>, or the next instance in a chain of clients.</param>
-    /// <param name="logger">An <see cref="ILogger"/> to use for logging information about function invocation.</param>
-    public FunctionInvokingChatClient(IChatClient innerClient, ILogger? logger = null)
+    /// <param name="loggerFactory">An <see cref="ILoggerFactory"/> to use for logging information about function invocation.</param>
+    /// <param name="services">An optional <see cref="IServiceProvider"/> to use for resolving services required by the <see cref="AIFunction"/> instances being invoked.</param>
+    public FunctionInvokingChatClient(IChatClient innerClient, ILoggerFactory? loggerFactory = null, IServiceProvider? services = null)
```

Comment:
I'm still a little worried that people will misunderstand the purpose IServiceProvider passed into the constructor and think it will get used by `FunctionInvokingChatClient.GetService(Type serviceType, object? serviceKey = null)`.

I suggested merging these concepts [here](https://github.com/dotnet/extensions/pull/6141#discussion_r2002047651), but I understand that might not be feasible given all of existing inner IChatClients don't expose arbitrary services that could be supplied by the host. Furthermore, the way IChatClients can wrap each other to add services wouldn't play very nicely with IServiceProviderIsService in the event we ever want to utilize that.

Maybe renaming the parameter will help avoid some of this confusion.

```suggestion
    public FunctionInvokingChatClient(IChatClient innerClient, ILoggerFactory? loggerFactory = null, IServiceProvider? functionInvocationServices = null)
```